### PR TITLE
Tower of hanoi added

### PR DESCRIPTION
**Pull request by shivammahajancse**
### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?

(Yes)

### Checklist

- [x] I've read the contribution guidelines.
- [x] I've checked the issue list before deciding what to submit.
- [x] I've edited the `README.md` and link to my code.

## Related Issues or Pull Requests

(#332 )

### CODE REVIEW

File: docs/data-structures/tower-of-hanoi.mdx [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+slug: /data-structures/tower-of-hanoi
+---
+
+## Introduction
+
+In this we are given three towers. Tower 1 has n disks placed one over the other in ascending order such that the smallest disk is at the top of Tower 1 similar to cone shape, where n is any positive number. Tower 2 and Tower 3 are empty. We have to move all discs from Tower 1 to Tower 3 using Tower 2.
+
+## Explanation 
+
+<img src="https://i.imgur.com/PCk9mpI.jpg?1" width="900" height="250"/>
```
Comment (atarax665) [inline docs/data-structures/tower-of-hanoi.mdx:12]: Make sure all the images used are your own creation.

File: docs/data-structures/tower-of-hanoi.mdx [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+We have to move all the disks from Tower 1 to Tower 3 (see in above figure) using Tower 2. 
+
+#### Properties:
+```
+If there are n number of discs, then we require atleast 2 raise to power n minus 1 moves.
+
+``` 
+
+#### Rules:
```
Comment (atarax665) [inline docs/data-structures/tower-of-hanoi.mdx:21]: I think the rules should be under ```Introduction``` heading itself.

File: docs/data-structures/tower-of-hanoi.mdx [hunk 3]
```diff
@@ -0,0 +1,161 @@
...
+Move disk 1 from rod A to rod C
+```
+
+## Complexity Analysis
+ 
+- Time Complexity: `O(2^n)`
+
+- Space Complexity: `O(n)`
+
+where `n` is the number of discs.
```
Comment (atarax665) [inline docs/data-structures/tower-of-hanoi.mdx:161]: Add credits.

**Review by atarax665 — CHANGES_REQUESTED** [review]

File: docs/data-structures/tower-of-hanoi.mdx (hunk 1 above)
Comment (shivammahajancse) [reply docs/data-structures/tower-of-hanoi.mdx:12]: Yes sir its created by me

File: docs/data-structures/tower-of-hanoi.mdx (hunk 3 above)
Comment (shivammahajancse) [reply docs/data-structures/tower-of-hanoi.mdx:161]: Code is written by me

File: docs/data-structures/tower-of-hanoi.mdx [hunk 4]
```diff
@@ -0,0 +1,161 @@
+---
+title: Tower of Hanoi
+slug: /data-structures/tower-of-hanoi
+---
+
+## Introduction
+
+In this we are given three towers. Tower 1 has n disks placed one over the other in ascending order such that the smallest disk is at the top of Tower 1 similar to cone shape, where n is any positive number. Tower 2 and Tower 3 are empty. We have to move all discs from Tower 1 to Tower 3 using Tower 2.
```
Comment (iamrajiv) [inline docs/data-structures/tower-of-hanoi.mdx:8]:
```suggestion
In this, we are given three towers. Tower 1 has n disks placed one over the other in ascending order such that the smallest disk is at the top of Tower 1 similar to a cone shape, where n is any positive number. Tower 2 and Tower 3 are empty. We have to move all discs from Tower 1 to Tower 3 using Tower 2.
```

File: docs/data-structures/tower-of-hanoi.mdx [hunk 5]
```diff
@@ -0,0 +1,161 @@
...
+
+## Explanation 
+
+<img src="https://i.imgur.com/PCk9mpI.jpg?1" width="900" height="250"/>
+We have to move all the disks from Tower 1 to Tower 3 (see in above figure) using Tower 2. 
+
+#### Properties:
+```
+If there are n number of discs, then we require atleast 2 raise to power n minus 1 moves.
```
Comment (iamrajiv) [inline docs/data-structures/tower-of-hanoi.mdx:17]:
```suggestion
If there are n number of discs, then we require at least 2 raised to power n minus 1 move.
```

File: docs/data-structures/tower-of-hanoi.mdx [hunk 6]
```diff
@@ -0,0 +1,161 @@
...
+In the above example we are given 3 towers A,B and C. 
+We have to move all the discs from tower A to tower C using tower B.
+Here we are given 3 discs, n=3.
+Disc 1 moved from A to C
+Disc 2 moved from A to B
+Disc 1 moved from C to B
+Disc 3 moved from A to C
+Disc 1 moved from B to A
+Disc 2 moved from B to C
+Disc 1 moved from A to C
```
Comment (iamrajiv) [inline docs/data-structures/tower-of-hanoi.mdx:46]:
```suggestion
In the above example, we are given 3 towers A, B and C. 
We have to move all the discs from tower A to tower C using tower B.
Here we are given 3 discs, n=3.
Disc 1 moved from A to C
Disc 2 moved from A to B
Disc 1 moved from C to B
Disc 3 moved from A to C
Disc 1 moved from B to A
Disc 2 moved from B to C
Disc 1 moved from A to C
```

File: docs/data-structures/tower-of-hanoi.mdx [hunk 7]
```diff
@@ -0,0 +1,161 @@
...
+
+```
+## Algorithm
+
+* If number of disc is 1, then simply move from source to destination.
```
Comment (iamrajiv) [inline docs/data-structures/tower-of-hanoi.mdx:51]:
```suggestion
* If the number of disc is 1, then simply move from source to destination.
```

**Review by iamrajiv — CHANGES_REQUESTED** [review]
@shivammahajancse Do the suggested changes.

**Comment by shivammahajancse** [comment]
All changes done sir @iamrajiv

**Review by iamrajiv — APPROVED** [review]

**Review by atarax665 — APPROVED** [review]
Everything else looks fine just relocate the file to maths folder and make the changes in sidebars.js

**Review by iamrajiv — CHANGES_REQUESTED** [review]
@shivammahajancse Resolve the merge conflict.

**Review by iamrajiv — APPROVED** [review]

**Pull request merged**